### PR TITLE
Added delimiter parameter to open functions of CsvReader and CsvWriter

### PR DESCRIPTION
**Pull request by jonathanudd**
Also added tests where `;` is used as a delimiter.

@thomasfanell peer review?

### CODE REVIEW

File: source/io/CsvReader.ooc [hunk 1]
```diff
@@ -12,7 +12,10 @@ import io/[File, FileReader]
 
 CsvReader: class extends Iterator<VectorList<Text>> {
 	_fileReader: FileReader
-	init: func (=_fileReader)
+	_delimiter: Char
```
Comment (marcusnaslund) [inline source/io/CsvReader.ooc]: I think having a read-only `delimiter` property could still be useful.

Comment (marcusnaslund) [inline source/io/CsvReader.ooc]: And likewise for the writer

File: test/io/CsvWriterTest.ooc [hunk 2]
```diff
@@ -45,6 +45,36 @@ CsvWriterTest: class extends Fixture {
 			outputFilename free()
 			csvRecords free()
 		})
+		this add("non default delimiter", func {
```
Comment (marcusnaslund) [inline test/io/CsvWriterTest.ooc]: `non-default`

**Comment by jonathanudd** [comment]
Updated

File: source/io/CsvReader.ooc [hunk 3]
```diff
@@ -12,7 +12,13 @@ import io/[File, FileReader]
...
+	delimiter: Char {
+		get { this _delimiter }
+	}
```
Comment (marcusnaslund) [inline source/io/CsvReader.ooc]: `delimiter ::= this _delimiter` is enough. :)

Comment (jonathanudd) [inline source/io/CsvReader.ooc]: Ok, will fix!

File: test/io/CsvReaderTest.ooc [hunk 4]
```diff
@@ -50,6 +50,24 @@ CsvReaderTest: class extends Fixture {
 			filename free()
 			reader free()
 		})
+		this add("non-default delimiter", func {
+			filename := Text new(c"test/io/input/semicolondelimiter.csv", 36)
```
Comment (thomasfanell) [inline test/io/CsvReaderTest.ooc]: is the `c` prefix necessary? `t"test/io/input/semicolondelimiter.csv"` should work, no?

Comment (thomasfanell) [inline test/io/CsvReaderTest.ooc]: Other than that, looks good

Comment (marcusnaslund) [inline test/io/CsvReaderTest.ooc]: `filename := t"test/io/input/semicolondelimiter.csv"`

Comment (jonathanudd) [inline test/io/CsvReaderTest.ooc]: Changed to Text literals everywhere in both CsvReader and CsvWriter tests.

**Comment by marcusnaslund** [comment]
Looks good to me too now.

**Pull request merged**